common/graalvm-reachability-metadata/src/main/java/org/graalvm/reachability/DirectoryConfiguration.java

```diff
@@ -60,4 +80,54 @@ public Path getDirectory() {
     public boolean isOverride() {
         return override;
     }
+
+    public static void copy(Collection<DirectoryConfiguration> configurations, Path destinationDirectory) throws IOException {
+        Path nativeImageDestination = destinationDirectory.resolve("META-INF").resolve("native-image");
+        Map<Path, DirectoryConfiguration> overrides = new LinkedHashMap<>();
+        for (DirectoryConfiguration configuration : configurations) {
+            Path target = nativeImageDestination
+                    .resolve(configuration.groupId)
+                    .resolve(configuration.artifactId)
+                    .resolve((configuration.version != null) ? configuration.version :
+                            configuration.getDirectory().getFileName().toString());
+            copyFileTree(configuration.directory, target);
+            if (configuration.isOverride()) {
+                overrides.put(target, configuration);
+            }
+        }
+        for (Map.Entry<Path, DirectoryConfiguration> entry : overrides.entrySet()) {
```

Comment:
As detailed on https://github.com/spring-projects/spring-boot/issues/31782#issuecomment-1266435509, I don't think `--exclude-config` is supported in `native-image.properties` so this part probably needs to be removed from that PR.

Reply:
I think we're going to need some way to discover if metadata had the `override` flag set so that we can reconstruct the exclude patterns later. The easiest way I can think of to do that is to write a specific properties file next to the metadata JSON. I'll update the PR to do that.